test/servinge2e/servicemesh_test.go

```diff
@@ -190,3 +231,295 @@ func TestKsvcWithServiceMeshSidecar(t *testing.T) {
 		})
 	}
 }
+
+// formats an RSA public key as JWKS
+func rsaPublicKeyAsJwks(key rsa.PublicKey, kid string) (string, error) {
+	eString := base64.RawURLEncoding.EncodeToString(big.NewInt(int64(key.E)).Bytes())
+	nString := base64.RawURLEncoding.EncodeToString(key.N.Bytes())
+	kidString := base64.RawURLEncoding.EncodeToString([]byte(kid))
```

Comment:
What about `keyIdString` and the function argument named as `keyId` ? I was first thinking whose KID is it? :)